Created Profanity Filter

ChrissprayCode opened the pull request:
Moved the blocked strings into their own file, Checking them against each word in the message.

Added bool options for 
A) Using the profanity filter at all
B) Blocking messages with any kind of upper/lowercase combo

ChrissprayCode committed:
created profanity filter

Uploaded text file for profanity and added a filter

valkyrienyanko commented on src/Server/Packets/Handle/HandlePacketChatMessage.cs:
```diff
@@ -38,10 +39,20 @@ public override void Handle(Event netEvent, ref PacketReader packetReader)
+            if(useProfanityFilter){
+            foreach (var blockedString in blockedStrings){
 
+                if(preventAllCaps){ message = message.ToLower();}
```
If `preventAllCaps` is set to `true` then all messages will be set to lowercase regardless if they send messages like

"Hello World" will become "hello world" <-- (undesired behavior)
"HELLO WORLD" will become "hello world"

ChrissprayCode replied:
This should be solved now!

valkyrienyanko commented on src/Server/Packets/Handle/HandlePacketChatMessage.cs:
```diff
@@ -15,14 +16,14 @@ public class HandlePacketChatMessage : HandlePacket
+        public bool preventAllCaps = false;
+
+        // read in bad words from file and split them into string array
+        private readonly static string blockedStringFile = File.ReadAllText(@"\Blocked-Strings-List");
```
I've never seen anyone use `@` symbol before, does this mean it's relative to the file this script is being executed in? I know if you removed the `@` it would be relative to `bin/Debug/net5.0`

ChrissprayCode replied:
The @ symbol just means "Interpret this string literally" so you cant use escape characters, makes it easier to read really.
For example:
Writing the path without it would look like - string filename = "c:\\documents\\files\\u0066.txt";
Whereas writing it with the symbols is - string filename = @"c:\documents\files\u0066.txt";
But both of these both do the exact same thing

valkyrienyanko commented on src/Server/Packets/Handle/Blocked-Strings-List.txt:
```diff
@@ -0,0 +1,1383 @@
+analannie
+analsex
+angie
+angry
```
lol

ChrissprayCode committed:
Edited caps prevention

Changed the chat filter so instead of it turning the entire message to lowercase, it checks string against bad word case-insensitively.

ChrissprayCode committed:
Changed chat filter slightly

Moved the second replace in an else statement so it won't always check the entire message again after it replaces when preventCapsCombo is true.

valkyrienyanko commented:
Thanks will merge when I start working on chat message packets again.
